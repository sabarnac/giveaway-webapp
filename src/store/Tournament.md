Usage:

```tsx
import Config from "./store/config/Config";
import Round from "./store/round/Round";
import Participant from "./store/round/match/participant/Participant";

const tournament: Tournament = new Tournament(Config.instance);
const firstRound: Round = tournament.firstRound; // Round("round-1")
const lastRound: Round = tournament.lastRound; // Round("round-x")
const rounds: Round[] = tournament.rounds; // [Round("round-1"), ..., Round("round-x")]
const rounds: Participant = tournament.winner; // Participant("winner")
```
