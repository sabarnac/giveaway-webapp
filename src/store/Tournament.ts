import { observable, computed, action } from "mobx";
import Participant from "./Participant";
import last from "lodash.last";
import Config from "../config/Config";
import Round from "./Round";

/**
 * Class containing the details of a tournament.
 */
export default class Tournament {
  @observable private _rounds: Round[];
  private static _PARTICIPANTS: Participant[] = Config.allParticipants;

  private _createRound = (participants: Participant[]): Round =>
    new Round(participants);

  private _getRounds = (): Round[] => {
    const rounds: Round[] = [this._createRound(Tournament._PARTICIPANTS)];
    while (!this.hasWinner) {
      rounds.push(this._createRound(this._latestRound.winners));
    }
    return rounds;
  };

  public constructor() {
    this._rounds = this._getRounds();
  }

  @computed private get _latestRound(): Round {
    return <Round>last(this._rounds);
  }

  @computed private get hasWinner(): boolean {
    return this._latestRound.winners.length === 1;
  }

  /**
   * Gets the list of rounds in the tournament.
   * @returns {Round[]} The rounds in the tournament.
   */
  public get rounds(): Round[] {
    return this._rounds;
  }

  /**
   * Gets the final winner of the tournament.
   * @returns {Participant} The final winner.
   */
  @computed public get winner(): Participant {
    return <Participant>this._latestRound.winners[0];
  }
}
