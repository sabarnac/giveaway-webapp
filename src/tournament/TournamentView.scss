.tournament {
    min-height: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background: #000000;
    h1 {
        color: #ffffff;
    }
}