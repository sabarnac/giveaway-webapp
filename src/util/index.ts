export let isDevEnvironment = (): boolean =>
  process.env.NODE_ENV === "development";

export let createAvatarImage = (identifier: string): string => {
  return `https://api.adorable.io/avatars/480/${identifier}`;
};

export let isInRange = (num: number, start: number, end: number): boolean =>
  num >= start && num <= end;
