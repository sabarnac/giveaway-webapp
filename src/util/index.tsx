import React, { ComponentType } from "react";
import { Dispatch, useState } from "react";
import Config from "../store/config/Config";
import { Redirect } from "react-router";
import { Observer } from "mobx-react";

/**
 * Returns if it is currently a dev environment.
 * @return {boolean} Whether it is currently a dev environment.
 */
export const isDevEnvironment = (): boolean =>
  process.env.NODE_ENV === "development";

/**
 * Returns a random avatar image URL.
 * @param {string} identifier The identifier to generate the avatar URL against.
 * @return {string} The avatar URL.
 */
export const createAvatarImage = (identifier: string): string => {
  return `https://api.adorable.io/avatars/480/${identifier}`;
};

/**
 * Checks if the number is within the range.
 * @param {number} num The number to check.
 * @param {number} start The start of the range.
 * @param {number} end The end of the range.
 * @return {boolean} Whether the number is within the range.
 */
export const isInRange = (num: number, start: number, end: number): boolean =>
  num >= start && num <= end;

/**
 * Type for custom animation state hook.
 */
export type AnimationStateHook = [
  number,
  () => void,
  (delay: number) => void,
  (state: number) => void
];

/**
 * Type for show overlay state hook.
 */
export type ShowOverlayHook = [boolean, (newValue: boolean) => void];

/**
 * Type for void function.
 */
export type VoidFunction = () => void;

/**
 * Runs the given action at least after the given delay.
 * @param {Function} action The action to run.
 * @param {number} delay The minimum delay to wait for.
 */
export const runOnDelay = (action: Function, delay: number): number =>
  setTimeout(action, delay);

/**
 * Create an animation state hook.
 * @return {AnimationStateHook} The current animation state, a state update method, and a delayed state update method.
 */
export const useAnimationState = (start: number = 0): AnimationStateHook => {
  const [currentState, setState]: [number, Dispatch<number>] = useState(start);

  const updateState = (): void => setState(currentState + 1);
  const updateStateDelay = (delay: number): VoidFunction => {
    const delayId: number = runOnDelay(updateState, delay);
    return () => clearTimeout(delayId);
  };

  return [currentState, updateState, updateStateDelay, setState];
};

/**
 * Normalize the given animation time by the animation speed multiplier.
 * @param {number} time The animation time to normalize.
 * @return {number} The normalized animation speed
 */
export const getNormalizedSpeed = (time: number): number =>
  time / Config.getInstance().speed;

/**
 * Runs the given action if the predicate is true.
 * @param {boolean} predicate The predicate to check.
 * @param {Function} action The action to run.
 */
export const runOnPredicate = (
  predicate: boolean,
  action: Function,
): VoidFunction | (() => VoidFunction) => (): void | VoidFunction =>
  predicate ? action() : undefined;

/**
 * Returns a redirect component for a given round and match.
 * @param {string} roundId The ID of the round to redirect to.
 * @param {string} matchId The ID of the match to redirect to.
 * @return {JSX.Element} The redirect element.
 */
export const getMatchRedirect = (roundId?: string, matchId?: string) => (
  <Redirect to={`/round/${roundId}/match/${matchId}`} />
);

/**
 * Returns a HOC with the given component wrapped by a MobX Observer component.
 * @param {ComponentType<T>} Component The component that should be under an observer.
 * @returns {(props: T) => JSX.Element} A HOC that wraps the given component under an observer.
 */
export const createObserver = <T extends {}>(
  Component: ComponentType<T>,
): ((props: T) => JSX.Element) => (props: T): JSX.Element => (
  <Observer>{() => <Component {...props} />}</Observer>
);
