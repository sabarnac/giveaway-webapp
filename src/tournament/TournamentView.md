Usage:

```tsx
import Config from "./store/config/Config";
import Tournament from "./store/Tournament";

const tournament: Tournament = new Tournament(Config.instance);
<TournamentView roundId="round-1" matchId="match-1" tournament={tournament} />;
```
