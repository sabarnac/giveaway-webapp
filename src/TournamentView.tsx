import React, { Component, Fragment } from "react";
import { observer } from "mobx-react";
import "./TournamentView.scss";
import classNames from "classnames";
import Tournament from "./store/Tournament";
import DevTools from "mobx-react-devtools";
import { isDevEnvironment } from "./util";
import ParticipantEntry from "./participant/entry/ParticipantEntry";

interface TournamentViewProps {
  tournament: Tournament;
}

@observer
class TournamentView extends Component<TournamentViewProps> {
  private getMobxDevTools = (): JSX.Element | null =>
    isDevEnvironment() ? <DevTools /> : null;

  public render = (): JSX.Element => {
    return (
      <Fragment>
        <div className={classNames("tournament")}>
          <ParticipantEntry participant={this.props.tournament.winner} />
        </div>
        {this.getMobxDevTools()}
      </Fragment>
    );
  };
}

export default TournamentView;
