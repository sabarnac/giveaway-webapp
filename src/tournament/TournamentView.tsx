import React, { Component, Fragment } from "react";
import { observer } from "mobx-react";
import "./TournamentView.scss";
import classNames from "classnames";
import Tournament from "../store/Tournament";
import DevTools from "mobx-react-devtools";
import { isDevEnvironment } from "../util";
import RoundView from "./round/RoundView";
import Round from "../store/round/Round";
import { Redirect } from "react-router";
import { isInRange } from "../util/index";
import { CSSTransition } from "react-transition-group";
import WinnerOverlay from "./winner-overlay/WinnerOverlay";
import LoserOverlay from "./loser-overlay/LoserOverlay";

interface TournamentViewProps {
  roundId: string;
  matchId: string;
  tournament: Tournament;
}

interface TournamentViewState {
  currentState: number;
  loserOverlayShown: boolean;
}

@observer
class TournamentView extends Component<
  TournamentViewProps,
  TournamentViewState
> {
  private _isMounted: boolean = false;

  private goToNextState = (): void =>
    this._isMounted
      ? this.setState({
          ...this.state,
          currentState: this.state.currentState + 1
        })
      : undefined;

  private toggleLoserOverlayShown = (): void =>
    this._isMounted
      ? this.setState({
          ...this.state,
          loserOverlayShown: !this.state.loserOverlayShown
        })
      : undefined;

  private getMobxDevTools = (): JSX.Element | null =>
    isDevEnvironment() ? <DevTools /> : null;

  private isCurrentRound = (round: Round): boolean =>
    round.id === this.props.roundId;

  private getCurrentRoundDetails = (): Round =>
    this.props.tournament.rounds.find(this.isCurrentRound) as Round;

  private getCurrentRoundIndex = (): number =>
    this.props.tournament.rounds.findIndex(this.isCurrentRound);

  private isLastRound = (): boolean =>
    this.props.tournament.lastRound.id === this.props.roundId;

  private getCurrentRound = (): JSX.Element => (
    <RoundView
      key={this.props.roundId}
      round={this.getCurrentRoundDetails()}
      matchId={this.props.matchId}
      onMatchComplete={this.goToNextState}
    />
  );

  private getTournamentDetails = (): JSX.Element => (
    <CSSTransition
      in={this.state.currentState > 0}
      timeout={500}
      classNames={{
        enter: "",
        enterActive: "round--entering",
        enterDone: "round--entered",
        exit: "",
        exitActive: "round--exiting",
        exitDone: "round--exited"
      }}
      mountOnEnter={true}
      unmountOnExit={true}
    >
      <div className={classNames("tournament")}>{this.getCurrentRound()}</div>
    </CSSTransition>
  );

  private getFirstRoundRedirect = (): JSX.Element | null =>
    this.getCurrentRoundIndex() === -1 ? (
      <Redirect to={`/round/${this.props.tournament.firstRound.id}`} />
    ) : null;

  private shouldAddNextRoundRedirect = (): boolean =>
    isInRange(
      this.state.currentState,
      0,
      (this.props.tournament.rounds.length - 1) * 2 + 1
    ) &&
    this.state.loserOverlayShown &&
    this.state.currentState % 2 === 0 &&
    this.props.tournament.rounds[this.state.currentState / 2].id !==
      this.props.roundId;

  private getNextRoundRedirect = (): JSX.Element | null =>
    this.shouldAddNextRoundRedirect() ? (
      <Redirect
        to={`/round/${
          this.props.tournament.rounds[this.state.currentState / 2].id
        }`}
      />
    ) : null;

  private shouldShowWinnerOverlay = (): boolean =>
    this.state.currentState >= (this.getCurrentRoundIndex() + 1) * 2 &&
    this.isLastRound();

  private getWinnerOverlay = (): JSX.Element => (
    <CSSTransition
      in={this.shouldShowWinnerOverlay()}
      timeout={500}
      classNames={{
        enter: "",
        enterActive: "round--entering",
        enterDone: "round--entered",
        exit: "",
        exitActive: "round--exiting",
        exitDone: "round--exited"
      }}
      mountOnEnter={true}
      unmountOnExit={true}
    >
      <WinnerOverlay winner={this.props.tournament.winner} />
    </CSSTransition>
  );

  private shouldShowLoserOverlay = (): boolean =>
    isInRange(
      this.state.currentState,
      0,
      (this.props.tournament.rounds.length - 1) * 2 + 1
    ) &&
    !this.state.loserOverlayShown &&
    this.state.currentState % 2 === 0 &&
    this.props.tournament.rounds[this.state.currentState / 2].id !==
      this.props.roundId &&
    !this.isLastRound();

  private getLoserOverlay = (): JSX.Element | null =>
    this.shouldShowLoserOverlay() && this.getCurrentRoundIndex() !== -1 ? (
      <LoserOverlay
        losers={this.getCurrentRoundDetails().losers}
        onOverlayComplete={() => this.toggleLoserOverlayShown()}
      />
    ) : null;

  public componentWillMount = (): void => {
    this.setState({
      loserOverlayShown: false,
      currentState:
        this.getCurrentRoundIndex() === -1 ? 0 : this.getCurrentRoundIndex() * 2
    });
  };

  public componentDidMount = (): void => {
    this._isMounted = true;
    this.goToNextState();
  };

  public componentDidUpdate = (prevProps: TournamentViewProps): void => {
    if (this.props.roundId !== prevProps.roundId) {
      this.setState({
        ...this.state,
        currentState: this.getCurrentRoundIndex() * 2 + 1
      });
    }
  };

  public componentWillUnmount = (): void => {
    this._isMounted = false;
  };

  public render = (): JSX.Element => {
    return (
      <Fragment key={`round-${this.getCurrentRoundIndex()}`}>
        {this.getTournamentDetails()}
        {this.getFirstRoundRedirect()}
        {this.getNextRoundRedirect()}
        {this.getLoserOverlay()}
        {this.getWinnerOverlay()}
        {this.getMobxDevTools()}
      </Fragment>
    );
  };
}

export default TournamentView;
