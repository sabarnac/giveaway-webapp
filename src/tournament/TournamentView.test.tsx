import React from "react";
import TournamentView from "./TournamentView";
import Tournament from "../store/Tournament";
import { create, ReactTestRenderer } from "react-test-renderer";
import { createDummyTournament } from "../util/test";

describe("TournamentView Component.", () => {
  it("Matches snapshot.", () => {
    const tournament: Tournament = createDummyTournament();
    const component: ReactTestRenderer = create(
      <TournamentView
        tournament={tournament}
        roundId={tournament.firstRound.id}
        matchId={tournament.firstRound.firstMatch.id}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
