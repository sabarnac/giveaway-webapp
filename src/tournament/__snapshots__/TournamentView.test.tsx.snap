// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TournamentView Component. Matches snapshot. 1`] = `
<div
  className="tournament"
>
  <h1>
    Foobar Giveaway
     Tournament
  </h1>
  <div
    className="speed-control"
  >
    <button
      className="speed-control__option button-primary"
      onClick={[Function]}
    >
      1x
    </button>
    <button
      className="speed-control__option"
      onClick={[Function]}
    >
      2x
    </button>
    <button
      className="speed-control__option"
      onClick={[Function]}
    >
      3x
    </button>
    <button
      className="speed-control__option"
      onClick={[Function]}
    >
      4x
    </button>
    <button
      className="speed-control__option"
      onClick={[Function]}
    >
      5x
    </button>
  </div>
  <div
    classNames={
      Object {
        "enter": "",
        "enterActive": "round--entering",
        "enterDone": "round--entered",
        "exit": "",
        "exitActive": "round--exiting",
        "exitDone": "round--exited",
      }
    }
    in={false}
    mountOnEnter={true}
    timeout={500}
    unmountOnExit={true}
  >
    <div
      classNames={
        Object {
          "enter": "",
          "enterActive": "round--entering",
          "enterDone": "round--entered",
          "exit": "",
          "exitActive": "round--exiting",
          "exitDone": "round--exited",
        }
      }
      in={false}
      mountOnEnter={true}
      timeout={500}
      unmountOnExit={true}
    >
      <div
        className="round"
        style={
          Object {
            "transition": "opacity 500ms ease-in-out",
          }
        }
      >
        <h2>
          Round 
          1
        </h2>
        <div
          className="round__list"
        >
          <div
            classNames={
              Object {
                "enter": "",
                "enterActive": "match--entering",
                "enterDone": "match--entered",
                "exit": "",
                "exitActive": "match--exiting",
                "exitDone": "match--exited",
              }
            }
            in={false}
            mountOnEnter={true}
            onEntering={[Function]}
            onExited={[Function]}
            timeout={500}
            unmountOnExit={true}
          >
            <div
              className="match"
              style={
                Object {
                  "transition": "opacity 500ms ease-in-out",
                }
              }
            >
              <div
                className="match__list"
              >
                <div
                  className="participant participant-entry"
                >
                  <div
                    className="participant-entry__avatar"
                  >
                    <img
                      alt="fozbaz-1"
                      src="https://api.adorable.io/avatars/480/fozbaz-1"
                    />
                  </div>
                  <div
                    className="participant-entry__name"
                  >
                    <strong>
                      Fozbaz 1
                    </strong>
                  </div>
                </div>
                <div
                  className="participant participant-entry"
                >
                  <div
                    className="participant-entry__avatar"
                  >
                    <img
                      alt="fozbaz-2"
                      src="https://api.adorable.io/avatars/480/fozbaz-2"
                    />
                  </div>
                  <div
                    className="participant-entry__name"
                  >
                    <strong>
                      Fozbaz 2
                    </strong>
                  </div>
                </div>
              </div>
              <div
                classNames={
                  Object {
                    "enter": "",
                    "enterActive": "match__interim--entering",
                    "enterDone": "match__interim--entered",
                    "exit": "",
                    "exitActive": "match__interim--exiting",
                    "exitDone": "match__interim--exited",
                  }
                }
                in={false}
                mountOnEnter={true}
                onEntered={[Function]}
                onExited={[Function]}
                timeout={200}
                unmountOnExit={true}
              >
                <div
                  className="match__interim"
                  style={
                    Object {
                      "transition": "opacity 200ms ease-in-out",
                    }
                  }
                >
                  <div
                    className="participant participant-entry"
                  >
                    <div
                      className="participant-entry__avatar"
                    >
                      <img
                        alt="fozbaz-1"
                        src="https://api.adorable.io/avatars/480/fozbaz-1"
                      />
                    </div>
                    <div
                      className="participant-entry__name"
                    >
                      <strong>
                        Fozbaz 1
                      </strong>
                    </div>
                  </div>
                </div>
              </div>
              <div
                classNames={
                  Object {
                    "enter": "",
                    "enterActive": "match__interim--entering",
                    "enterDone": "match__interim--entered",
                    "exit": "",
                    "exitActive": "match__interim--exiting",
                    "exitDone": "match__interim--exited",
                  }
                }
                in={false}
                mountOnEnter={true}
                onEntered={[Function]}
                onExited={[Function]}
                timeout={200}
                unmountOnExit={true}
              >
                <div
                  className="match__interim"
                  style={
                    Object {
                      "transition": "opacity 200ms ease-in-out",
                    }
                  }
                >
                  <div
                    className="participant participant-entry"
                  >
                    <div
                      className="participant-entry__avatar"
                    >
                      <img
                        alt="fozbaz-2"
                        src="https://api.adorable.io/avatars/480/fozbaz-2"
                      />
                    </div>
                    <div
                      className="participant-entry__name"
                    >
                      <strong>
                        Fozbaz 2
                      </strong>
                    </div>
                  </div>
                </div>
              </div>
              <div
                classNames={
                  Object {
                    "enter": "",
                    "enterActive": "match-overlay-wrapper--entering",
                    "enterDone": "match-overlay-wrapper--entered",
                    "exit": "",
                    "exitActive": "match-overlay-wrapper--exiting",
                    "exitDone": "match-overlay-wrapper--exited",
                  }
                }
                in={false}
                mountOnEnter={true}
                onExited={[Function]}
                timeout={500}
                unmountOnExit={true}
              >
                <div
                  className="match-overlay-wrapper"
                  style={
                    Object {
                      "transition": "opacity 500ms ease-in-out",
                    }
                  }
                >
                  <div
                    className="match-overlay"
                  >
                    <div
                      className="match-overlay__list"
                    >
                      <div>
                        <div
                          className="participant participant-card participant--invert"
                        >
                          <div
                            className="participant-card__avatar"
                          >
                            <img
                              alt="fozbaz-1"
                              src="https://api.adorable.io/avatars/480/fozbaz-1"
                            />
                          </div>
                          <div
                            className="participant-card__name"
                          >
                            <strong>
                              Fozbaz 1
                            </strong>
                          </div>
                        </div>
                      </div>
                      <h3
                        className="versus-text"
                      >
                        VS
                      </h3>
                      <div>
                        <div
                          className="participant participant-card participant--invert"
                        >
                          <div
                            className="participant-card__avatar"
                          >
                            <img
                              alt="fozbaz-2"
                              src="https://api.adorable.io/avatars/480/fozbaz-2"
                            />
                          </div>
                          <div
                            className="participant-card__name"
                          >
                            <strong>
                              Fozbaz 2
                            </strong>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      classNames={
                        Object {
                          "enter": "",
                          "enterActive": "match-overlay__interim--entering",
                          "enterDone": "match-overlay__interim--entered",
                          "exit": "",
                          "exitActive": "match-overlay__interim--exiting",
                          "exitDone": "match-overlay__interim--exited",
                        }
                      }
                      in={false}
                      mountOnEnter={true}
                      onEntered={[Function]}
                      onExited={[Function]}
                      timeout={500}
                      unmountOnExit={true}
                    >
                      <div
                        className="match-overlay__interim"
                        style={
                          Object {
                            "transition": "opacity 500ms ease-in-out",
                          }
                        }
                      >
                        <h3>
                          Selecting Winner
                        </h3>
                        <div
                          className="css-yd6m5-style"
                        />
                      </div>
                    </div>
                    <div
                      classNames={
                        Object {
                          "enter": "",
                          "enterActive": "match-overlay__winner--entering",
                          "enterDone": "match-overlay__winner--entered",
                          "exit": "",
                          "exitActive": "match-overlay__winner--exiting",
                          "exitDone": "match-overlay__winner--exited",
                        }
                      }
                      in={false}
                      mountOnEnter={true}
                      onEntered={[Function]}
                      timeout={500}
                      unmountOnExit={true}
                    >
                      <div
                        className="match-overlay__winner"
                        style={
                          Object {
                            "transition": "opacity 500ms ease-in-out",
                          }
                        }
                      >
                        <div
                          className="participant participant-entry"
                        >
                          <div
                            className="participant-entry__avatar"
                          >
                            <img
                              alt="fozbaz-1"
                              src="https://api.adorable.io/avatars/480/fozbaz-1"
                            />
                          </div>
                          <div
                            className="participant-entry__name"
                          >
                            <strong>
                              Fozbaz 1
                            </strong>
                          </div>
                        </div>
                        <h3>
                          Won The Match!
                        </h3>
                        <h5>
                          <strong>
                            <em>
                              formatted foobar message
                            </em>
                          </strong>
                        </h5>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    classNames={
      Object {
        "enter": "",
        "enterActive": "winner-overlay-wrapper--entering",
        "enterDone": "winner-overlay-wrapper--entered",
        "exit": "",
        "exitActive": "winner-overlay-wrapper--exiting",
        "exitDone": "winner-overlay-wrapper--exited",
      }
    }
    in={false}
    mountOnEnter={true}
    timeout={500}
    unmountOnExit={true}
  >
    <div
      className="winner-overlay-wrapper"
      style={
        Object {
          "transition": "opacity 500ms ease-in-out",
        }
      }
    >
      <div
        className="winner-overlay"
      >
        <div
          className="winner-overlay__winner"
        >
          <div
            className="participant participant-card"
          >
            <div
              className="participant-card__avatar"
            >
              <img
                alt="fozbaz-1"
                src="https://api.adorable.io/avatars/480/fozbaz-1"
              />
            </div>
            <div
              className="participant-card__name"
            >
              <strong>
                Fozbaz 1
              </strong>
            </div>
          </div>
          <h3>
            Won The Giveaway!
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
